[runtime] Fix FromDLPackCapsule creating buffer with access ANY

devtbi opened the pull request:
`iree_hal_buffer_validate_access` did not check if the `allowed_memory_access` was set to `IREE_HAL_MEMORY_ACCESS_ANY`. It therefore failed e.g. if `IREE_HAL_MEMORY_ACCESS_READ` was required and `IREE_HAL_MEMORY_ACCESS_ANY` was allowed. According to the documentation, `ANY` "may perform any operation and should not be validated". This commit adds the missing check.

benvanik reviewed (changes requested):
IREE_HAL_MEMORY_ACCESS_ANY isn't allowed (or at least, not intended) to be used like this - it's supposed to only be used on accesses. If a buffer wants to allow any access type it should specify them explicitly or use IREE_HAL_MEMORY_ACCESS_ALL. Is there a place you found that's incorrectly passing it?

(new CTS tests are always welcome! but it should be allocating with IREE_HAL_MEMORY_ACCESS_ALL)

devtbi committed:
[runtime] Fix FromDLPackCapsule creating buffer with access ANY

ANY should not be used for the allowed access. Change the allowed access to READ | WRITE.

Signed-off-by: Tobias <fuchs@roofline.ai>
